module/zfs/zpl_file.c

```diff
@@ -798,7 +801,17 @@ zpl_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 static long
 zpl_compat_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 {
-	return (zpl_ioctl(filp, cmd, arg));
+	switch (cmd) {
+	case FS_IOC32_GETFLAGS:
+		cmd = FS_IOC_GETFLAGS;
+		break;
+	case FS_IOC32_SETFLAGS:
+		cmd = FS_IOC_SETFLAGS;
+		break;
+	default:
+		return (-ENOTTY);
```

Comment:
autofs4, btrfs, ceph, cifs, hfsplus, jffs2, jffs, xfs return -ENOTTY,  ncpfs returns -EINVAL, and ubifs returns -ENOTTY and also -ENOIOCTLCMD.  The rest return --ENOIOCTLCMD.  So, I'm not sure what the correct error return is.  According to ioctl(2) manual, -ENOTTY seems like the documented return for a command that is not recognised for the specific object being operated on.